smart:
# controls the rear two tires based on individual motors
smart:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

  # Velocity Controllers ---------------------------------------
  rear_left_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: back_left_joint
    pid: {p: 10.0, i: 0.01, d: 0.0}
  rear_right_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: back_right_joint
    pid: {p: 10.0, i: 0.01, d: 0.0}
  
  front_left_steering_position_controller:
    type: effort_controllers/JointPositionController
    joint: front_left_steer_joint
    pid: {p: 1.0, i: 0.01, d: 0.0}
  front_right_steering_position_controller:
    type: effort_controllers/JointPositionController
    joint: front_right_steer_joint
    pid: {p: 1.0, i: 0.01, d: 0.0}

  # gazebo_ros_control:
  #   pid_gains:
  #     rear_right_wheel_joint:
  #       p: 1000.0
  #       i: 0.5
  #       d: 0.0
  #     rear_left_wheel_joint:
  #       p: 1000.0
  #       i: 0.5
  #       d: 0.0